invenio_rdm_records/requests/community_inclusion.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Community addition request."""
+
+from invenio_i18n import lazy_gettext as _
+from invenio_records_resources.services.uow import RecordCommitOp, RecordIndexOp
+from invenio_requests.customizations import RequestType, actions
+
+from ..proxies import current_rdm_records_service as service
+
+
+#
+# Actions
+#
+class SubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        # Set the record's title as the request title.
+        self.request["title"] = record.metadata["title"]
+        super().execute(identity, uow)
+
+
+class AcceptAction(actions.AcceptAction):
+    """Accept action."""
+
+    def execute(self, identity, uow):
+        """Include record into community."""
+        # Resolve the topic and community - the request type only allow for
+        # community receivers and record topics.
+        record = self.request.topic.resolve()
+        community = self.request.receiver.resolve()
+
+        # integrity check, it should never happen on a published record
+        assert not record.parent.review
+
+        # set the community to `default` if it is the first
+        default = not record.parent.communities
```

Comment:
should we start a ticket to address this as changeable in the future?

Reply:
Yes if you prefer: it will require discussions and mock-ups on how to make this easy for the users.